specification/cognitiveservices/data-plane/Face/preview/v1.0-preview/Face.json

```diff
@@ -0,0 +1,5548 @@
+        "operationId": "Snapshot_Apply",
+        "parameters": [
+          {
+            "$ref": "#/parameters/snapshotId"
+          },
+          {
+            "name": "body",
+            "description": "Request body for applying a snapshot.",
+            "in": "body",
+            "required": true,
+            "x-ms-client-flatten": true,
+            "schema": {
+              "$ref": "#/definitions/ApplySnapshotRequest"
+            }
+          }
+        ],
+        "consumes": [
+          "application/json"
+        ],
+        "produces": [
+          "application/json"
+        ],
+        "responses": {
+          "202": {
+            "description": "The snapshot applying task was queued successfully.",
+            "headers": {
+              "Operation-Location": {
+                "description": "Operation location with an operation id used to track the progress of applying the snapshot by OperationStatus - Get.",
+                "type": "string"
+              }
+            }
+          },
+          "default": {
+            "description": "Error response.",
+            "schema": {
+              "$ref": "#/definitions/APIError"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Apply snapshot example": {
+            "$ref": "./examples/ApplySnapshot.json"
+          }
+        }
+      }
+    },
+    "/operations/{operationId}": {
+      "get": {
+        "description": "Get status of a long running operation.",
+        "operationId": "Snapshot_GetOperationStatus",
+        "parameters": [
+          {
+            "$ref": "#/parameters/operationId"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "A successful call returns the operation's status.",
+            "schema": {
+              "$ref": "#/definitions/AsyncStatus"
```

Comment:
I see this changed from v1.0.  Does this need the [x-ms-long-running-operation](https://github.com/Azure/autorest/tree/master/docs/extensions#x-ms-long-running-operation) annotation?

Reply:
The updated schema is still compatible with the previous version, just with some new fields.
Regarding the x-ms-long-running-operation annotation, it probably would have made sense to be added to the operations that actually produce 202s, but given that FaceAPI has been following the [API guidelines to use Operation-Location](https://github.com/Microsoft/api-guidelines/blob/master/Guidelines.md#1321-put) in the header, it seems incompatible with the final-state-via definitions. For the sake of consistency with existing long-running operations in the service, we would prefer to keep it the current way for now.